ibm_cloud_sdk_core/iam_token_manager.py

```diff
@@ -77,10 +79,15 @@ def __init__(self,
                  proxies: Optional[Dict[str, str]] = None,
                  scope: Optional[str] = None) -> None:
         self.apikey = apikey
-        self.url = url if url else self.DEFAULT_IAM_URL
+        if not url:
+            url = self.DEFAULT_IAM_URL
+        if url.endswith(self.OPERATION_PATH):
+            url = url[:-len(self.OPERATION_PATH)]
+        self.url = url
```

Comment:
note: at this point, `self.url` should be a valid IAM base URL string without the path part.